### PR TITLE
[cupertino_http] A bit of refactoring of the example

### PR DESCRIPTION
**Pull request by mit-mit**
- Give a bit of user feedback for case where there is no search query
- Use async/await per Dart style
- Split code across a few files
- Improve indentation a bit

### CODE REVIEW

File: pkgs/cupertino_http/example/lib/get_books.dart [hunk 1]
```diff
@@ -0,0 +1,38 @@
+import 'dart:convert';
+import 'package:http/http.dart';
+
+import 'book.dart';
+
+Future<List<Book>> getBooks(String query) async {
+  final response = await get(
```
Comment (brianquinlan) [inline pkgs/cupertino_http/example/lib/get_books.dart:7]: You don't think that it is valuable so have this comment?

// `get` will use the `Client` configured in main.

File: pkgs/cupertino_http/example/lib/get_books.dart [hunk 2]
```diff
@@ -0,0 +1,38 @@
+import 'dart:convert';
```
Comment (brianquinlan) [inline pkgs/cupertino_http/example/lib/get_books.dart:1]: Don't you only see `main.dart` on pub (i.e. https://pub.dev/packages/cupertino_http/example) so the network code will disappear from the example.

File: pkgs/cupertino_http/example/lib/get_books.dart (hunk 1 above)
Comment (mit-mit) [reply pkgs/cupertino_http/example/lib/get_books.dart]: Yes, good idea, added

File: pkgs/cupertino_http/example/lib/get_books.dart (hunk 2 above)
Comment (mit-mit) [reply pkgs/cupertino_http/example/lib/get_books.dart]: Good point, moved the JSON code to book.dart and then the networking code back to main.dart.

**Comment by mit-mit** [comment]
Updated, PTAL @brianquinlan

File: pkgs/cupertino_http/example/lib/main.dart [hunk 3]
```diff
@@ -49,88 +45,71 @@ class HomePage extends StatefulWidget {
...
+    final response = await get(
+      Uri.https(
+        'www.googleapis.com',
+        '/books/v1/volumes',
+        {'q': query, 'maxResults': '40', 'printType': 'books'},
+      ),
+    );
+
+    final json = jsonDecode(utf8.decode(response.bodyBytes)) as Map;
+    return Book.listFromJson(json);
```
Comment (brianquinlan) [inline pkgs/cupertino_http/example/lib/main.dart:67]: Nice! I was annoying how much tedious parsing was getting in the way of my example ;-)

File: pkgs/cupertino_http/example/lib/book.dart [hunk 4]
```diff
@@ -0,0 +1,32 @@
+class Book {
```
[the remark this answers is not in the document]
Comment (mit-mit) [reply pkgs/cupertino_http/example/lib/book.dart:5]: Yes!

**Pull request merged**